history/package/marathon.json.mustache

```diff
@@ -2,27 +2,60 @@
     "id": "{{service.name}}",
     "cpus": {{service.cpus}},
     "mem": {{service.mem}},
+{{^security.kerberos.principal}}
     "cmd": "SPARK_HISTORY_OPTS=\"-Dspark.history.ui.port=${PORT0} ${SPARK_HISTORY_OPTS}\" ./bin/spark-class org.apache.spark.deploy.history.HistoryServer",
+{{/security.kerberos.principal}}
+{{#security.kerberos.principal}}
+    "cmd": "SPARK_HISTORY_OPTS=\"-Dspark.history.ui.port=${PORT0} ${SPARK_HISTORY_OPTS} ${SPARK_HISTORY_KERBEROS_OPTS}\" ./bin/spark-class org.apache.spark.deploy.history.HistoryServer",
+{{/security.kerberos.principal}}
+    "user": "{{service.user}}",
     "env": {
+{{#security.kerberos.krb5conf}}
+        "KRB5_CONFIG_BASE64": "{{security.kerberos.krb5conf}}",
+{{/security.kerberos.krb5conf}}
         "SPARK_USER": "{{service.user}}",
         "APPLICATION_WEB_PROXY_BASE": "/service/{{service.name}}",
+{{#security.kerberos.principal}}
+        "SPARK_HISTORY_KERBEROS_OPTS": "-Dspark.history.kerberos.enabled=true -Dspark.history.kerberos.principal={{security.kerberos.principal}} -Dspark.history.kerberos.keytab=/opt/spark/hdfs.keytab",
+{{/security.kerberos.principal}}
         "SPARK_HISTORY_OPTS": "-Dspark.history.fs.logDirectory={{service.log-dir}} -Dspark.history.fs.cleaner.enabled={{service.cleaner.enabled}} -Dspark.history.fs.cleaner.interval={{service.cleaner.interval}} -Dspark.history.fs.cleaner.maxAge={{service.cleaner.max-age}}"
     },
     "ports": [0],
     "container": {
-        "type": "DOCKER",
+        "type": "MESOS",
         "docker": {
             "image": "{{service.docker-image}}",
-            "network": "HOST",
-            "forcePullImage": true,
-            "parameters": [
-               {
-                 "key": "user",
-                 "value": "{{service.user}}"
-               }
-            ]
+            "forcePullImage": true
         }
+{{#security.kerberos.keytab}}
+        ,
+        "volumes": [
+            {
+                "containerPath": "/opt/spark/hdfs.keytab",
+                "secret": "hdfs_keytab",
```

Comment:
shouldn't this be templated? so that the user can use any keytab secret they want?